### ISSUE TITLE
latest ncremap failed with vertical interpolation

### ISSUE THREAD

**Issue by czender** [nco/nco#216]
Submitted to E3SM-Project/e3sm-unified#79 by @wlin7 /cc @xylar @whannah1

This happened on cori with latest e3sm_unified. The vertical interpolation was for an EAM initial file. The error message is
/global/cfs/cdirs/e3sm/software/anaconda_envs/base/envs/e3sm_unified_1.3.1/bin/ncremap: line 2767: 60854 Segmentation fault

The command to run and reproduce the error

cd /global/cscratch1/sd/wlin/share/testVrtInt
./map_ne30np4_to_ne120np4_vert_L72.sh.  # contains the actual ncremap cmd and options
The script used to work well. Instead of using laetest e3sm_unified, if activating e3sm_unified_1.3.0, it would still work.

This is not urgent since I can stick with 1.3.0 for now. Thanks,

**Comment by czender** [comment]
I can reproduce this with the latest snapshot, although the untimely death is now marked by a more useful error message that offers a clue as to how I must have broken the once healthy code between 4.9.1 and 4.9.2:

```
zender@cori08:~$` nice time ncremap --dbg=1 --devnull=No --vrt_fl=/global/cscratch1/sd/wlin/share/testVrtInt/E3SM_vert_coor_L72_ne120np4.nc /global/cscratch1/sd/wlin/share/testVrtInt/map_to_ne120np4.20200819.DYAMOND-2.spinup.ne30-r0125.cam.i.2019-12-01-00000.nc ~/data/foo.nc
Input #00: /global/cscratch1/sd/wlin/share/testVrtInt/map_to_ne120np4.20200819.DYAMOND-2.spinup.ne30-r0125.cam.i.2019-12-01-00000.nc
Vertical : /global/cscratch1/sd/wlin/share/testVrtInt/E3SM_vert_coor_L72_ne120np4.nc
ncks -O -t 2 --no_tmp_fl --hdr_pad=10000 --gaa remap_script=ncremap --gaa remap_hostname=cori08 --gaa remap_version=4.9.4-alpha15 --vrt="/global/cscratch1/sd/wlin/share/testVrtInt/E3SM_vert_coor_L72_ne120np4.nc" "/global/cscratch1/sd/wlin/share/testVrtInt/map_to_ne120np4.20200819.DYAMOND-2.spinup.ne30-r0125.cam.i.2019-12-01-00000.nc" "/global/homes/z/zender/data/foo.nc"
ncks: ERROR nco_ntp_vrt() reports that temporal or horizontal spatial dimensions differ: grd_sz_in = 1 != 777602 = grd_sz_out, tm_nbr_in = 1 != 1 = tm_nbr_out
ncks: nco_rgr.c:1299: nco_ntp_vrt: Assertion `grd_sz_in == grd_sz_out' failed.

```

**Comment by czender** [comment]
Thank you for reporting this. This turned out to be a one-line fix to an error I probably did introduce while refactoring the code for 4.9.2 to handle Walter's issue. The latest snapshot, NCO 4.9.4-alpha15, now on Cori and Compy, contains the fix which runs fine on Cori (below). If you have a chance to test further that would be great. Otherwise this fix will be in 4.9.4-final to be released in a week or so. There is a workaround that will allow 4.9.2 and 4.9.3 to run this command: move the PS variable from the vertical gridfile to the input datafile. If you have any more problems with vertical interpolation please re-open this issue.

```
zender@cori08:~/nco$ nice time ncremap --dbg=1 --devnull=No --vrt_fl=/global/cscratch1/sd/wlin/share/testVrtInt/E3SM_vert_coor_L72_ne120np4.nc /global/cscratch1/sd/wlin/share/testVrtInt/map_to_ne120np4.20200819.DYAMOND-2.spinup.ne30-r0125.cam.i.2019-12-01-00000.nc ~/data/foo.nc
Input #00: /global/cscratch1/sd/wlin/share/testVrtInt/map_to_ne120np4.20200819.DYAMOND-2.spinup.ne30-r0125.cam.i.2019-12-01-00000.nc
Vertical : /global/cscratch1/sd/wlin/share/testVrtInt/E3SM_vert_coor_L72_ne120np4.nc
ncks -O -t 2 --no_tmp_fl --hdr_pad=10000 --gaa remap_script=ncremap --gaa remap_hostname=cori08 --gaa remap_version=4.9.4-alpha15 --vrt="/global/cscratch1/sd/wlin/share/testVrtInt/E3SM_vert_coor_L72_ne120np4.nc" "/global/cscratch1/sd/wlin/share/testVrtInt/map_to_ne120np4.20200819.DYAMOND-2.spinup.ne30-r0125.cam.i.2019-12-01-00000.nc" "/global/homes/z/zender/data/foo.nc"
ncks: INFO nco_ntp_vrt() temperature extrapolated toward/into surface assuming constant moist adiabatic lapse rate = 6.5 K/(100 mb) for variable T
117.32user 36.37system 1:33.13elapsed 165%CPU (0avgtext+0avgdata 2739512maxresident)k
38031490inputs+0outputs (1major+10196665minor)pagefaults 0swaps
```
